packages/open-next/src/adapters/cacheIntercept.ts

```diff
@@ -0,0 +1,208 @@
+import { PrerenderManifest, revalidateInBackground } from "./util.js";
+import { GetObjectCommand, S3Client } from "@aws-sdk/client-s3";
+import { request as httpsRequest } from "https";
+import { debug } from "./logger.js";
+
+interface MatchedRoute {
+  key: string;
+  route: string;
+  // srcRoute: string;
+  revalidate: number;
+}
+
+interface CompiledDynamicRoute {
+  routeRegex: RegExp;
+  dataRouteRegex: RegExp;
+  srcRoute: string;
+}
+
+interface ProxyEvent {
+  rawPath: string;
+  headers: Record<string, string>;
+  method: string;
+  domainName: string;
+}
+
+interface ProxyResult {
+  statusCode: number;
+  headers: Record<string, string | string[]>;
+  body: string;
+  isBase64Encoded: boolean;
+}
+
+interface TypeMatcher {
+  html: string;
+  json: string;
+  rsc: string;
+}
+
+const DEFAULT_REVALIDATE = 365 * 24 * 60 * 60;
+
+export class CacheInterceptor {
```

Comment:
Actually this is optional to make ISR works. The only point of this class is to avoid having to spawn NextServer which could be pretty slow on cold start especially if you have some heavy libs in _app.js

Reply:
I see, that is more interesting.